File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for(int i = 0; i<nums.size(); i++)
        {
            seen[nums[i]]=i; 
        } 
        for(int i = 0; i < nums.size() -1; i++)
        {
            int j = target - nums[i];
            if(seen[j] == i)
            {

            } else if (seen[j]>0) {
                std::vector<int> ans(2);
                ans[0]=i;
                ans[1]=seen[j];
                return ans;
            }
            
        }
        return {};
        
        
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-3.py
class Solution:
    
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        self.answer=[]
        self.help(nums, target, [], 0, 0)
        return self.answer
        
    def help(self, nums: List[int], target: int, current: List[int], index: int, sum: int):
        if index == len(nums):
            return
        if sum == target:
            if not current in self.answer:
                self.answer.append(current.copy())
                return
        if sum>target:
            return
        if sum < target:
            self.help(nums, target, current, index+1, sum)
            current.append(nums[index])
            sum = sum + nums[index]
            self.help(nums, target, current, index, sum)
            current.pop()
        return


        